includes/embeds/class-amp-core-block-handler.php

```diff
@@ -178,4 +180,75 @@ public function ampify_cover_block( $block_content, $block ) {
 		}
 		return $block_content;
 	}
+
+	/**
+	 * Sanitize widgets that are not added via Gutenberg.
+	 *
+	 * @param Document $dom Document.
+	 */
+	public function sanitize_raw_embeds( Document $dom ) {
+		$this->process_categories_widgets( $dom );
+		$this->process_archives_widgets( $dom );
+	}
+
+	/**
+	 * Process "Categories" widgets.
+	 *
+	 * @since 1.5.2
+	 *
+	 * @param Document $dom Document.
+	 */
+	private function process_categories_widgets( Document $dom ) {
+		static $count = 0;
+
+		/**
+		 * Element.
+		 *
+		 * @var DOMElement $widget_container
+		 */
+
+		$widget_containers = $dom->xpath->query( '//*[ contains( @class, "widget_categories" ) ]' );
+		foreach ( $widget_containers as $widget_container ) {
+			$select = $dom->xpath->query( './/select[ @name = "cat" ]', $widget_container )->item( 0 );
+			$form   = $widget_container->getElementsByTagName( 'form' )->item( 0 );
+			if ( ! $select instanceof DOMElement || ! $form instanceof DOMElement ) {
+				continue;
+			}
+
+			$count++;
+			$id = sprintf( 'amp-wp-widget-categories-%d', $count );
+			$form->setAttribute( 'id', $id );
+
+			$select->setAttribute( 'on', sprintf( 'change:%s.submit', $id ) );
+			$script = $dom->xpath->query( './/script[ contains( text(), "onCatChange" ) ]', $widget_container )->item( 0 );
+			if ( $script ) {
+				$script->parentNode->removeChild( $script );
+			}
+		}
+	}
+
+	/**
+	 * Process "Archives" widgets.
+	 *
+	 * @since 1.5.2
+	 *
+	 * @param Document $dom Select node retrieved from the widget.
+	 */
+	private function process_archives_widgets( Document $dom ) {
+
+		$widget_containers = $dom->xpath->query( '//*[ contains( @class, "widget_archive" ) ]' );
+		foreach ( $widget_containers as $widget_container ) {
+			$select = $dom->xpath->query( './/select[ @name = "archive-dropdown" ]', $widget_container )->item( 0 );
+			if ( ! $select instanceof DOMElement ) {
+				continue;
+			}
+
+			$select->setAttribute( 'on', 'change:AMP.navigateTo(url=event.value)' );
```

Comment:
Use the `add_amp_action()` helper here to ensure a potential previous action is not just overwritten.
```suggestion
			AMP_DOM_Utils::add_amp_action( $select, 'change', 'AMP.navigateTo(url=event.value)' );
```